tests/ci/git_helper.py

```diff
@@ -122,17 +124,35 @@ class Git:
     _tag_pattern = re.compile(TAG_REGEXP)
 
     def __init__(self, ignore_no_tags: bool = False):
+        """
+        new_tag is used for special v24.1.1.1-new tags where the previous version is moved to the release branch
+        * 66666666666 Some another commit with version 24.8.1.xxxxx-testing, tweak is counted from new_tag = v24.8.1.1-new
+        | * 55555555555 (tag: v24.7.1.123123123-stable, branch: 24.7) tweak counted from new_tag = v24.7.1.1-new
+        |/
+        * 44444444444 (tag: v24.8.1.1-new)
```

Comment:
it's a bit counterintuitive in my opinion that the new release branch started from a commit with a tag with major versions for the next release:
v24.7.1.123123123-stable comes after tag v24.8.1.1-new

Reply:
the whole thing here and all other PRs linked to https://github.com/ClickHouse/ClickHouse/issues/64671 is because `v24.7.1.123123123-stable` used to be in the master. And now it's not `¯\_(ツ)_/¯`